File: Legacy/Objective-C/Neocom/Neocom/NCTableViewBackgroundLabel.m
//
//  NCTableViewBackgroundLabel.m
//  Neocom
//
//  Created by Artem Shimanski on 18.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

#import "NCTableViewBackgroundLabel.h"

@interface NCLabel()
@property (nonatomic, assign) CGFloat pointSize;
- (CGFloat) fontSizeWithContentSizeCategory:(UIContentSizeCategory) contentSizeCategory;
@end

@implementation NCTableViewBackgroundLabel

+ (instancetype) labelWithText:(NSString*) text {
	return [[self alloc] initWithText:text];
}

- (id) initWithFrame:(CGRect)frame {
	if (self = [super initWithFrame:frame]) {
		self.pointSize = 15;
		self.numberOfLines = 0;
		self.font = [UIFont systemFontOfSize:[self fontSizeWithContentSizeCategory:[[UIApplication sharedApplication] preferredContentSizeCategory]]];
		self.textColor = [UIColor lightTextColor];
	}
	return self;
}


- (instancetype) initWithText:(NSString*) text {
	if (self = [self initWithFrame:CGRectZero]) {
		NSMutableParagraphStyle* paragraph = [[NSParagraphStyle defaultParagraphStyle] mutableCopy];
		paragraph.firstLineHeadIndent = 20;
		paragraph.headIndent = 20;
		paragraph.tailIndent = -20;
		paragraph.alignment = NSTextAlignmentCenter;
		self.attributedText = [[NSAttributedString alloc] initWithString:text attributes:@{NSParagraphStyleAttributeName:paragraph}];
	}
	return self;
}

@end

File: Legacy/Neocom/Neocom/NCTableViewBackgroundLabel.swift
//
//  File.swift
//  Neocom
//
//  Created by Artem Shimanski on 08.12.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit

class NCTableViewBackgroundLabel: NCLabel {
	
	override init(frame: CGRect) {
		super.init(frame: frame)
		numberOfLines = 0
		pointSize = 15
		font = UIFont.systemFont(ofSize: fontSize(contentSizeCategory: UIApplication.shared.preferredContentSizeCategory))
		textColor = UIColor.lightText
	}
	
	convenience init(text: String) {
		self.init(frame: CGRect.zero)
		let paragraph = NSParagraphStyle.default.mutableCopy() as! NSMutableParagraphStyle
		paragraph.firstLineHeadIndent = 20
		paragraph.headIndent = 20
		paragraph.tailIndent = -20
		paragraph.alignment = NSTextAlignment.center
		attributedText = NSAttributedString(string: text, attributes: [NSAttributedStringKey.paragraphStyle: paragraph])
	}
	
	required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
}
